mesonbuild/coredata.py

```diff
@@ -91,7 +91,8 @@ class MesonVersionMismatchException(MesonException):
     '''Build directory generated with Meson version is incompatible with current version'''
     def __init__(self, old_version: str, current_version: str) -> None:
         super().__init__(f'Build directory has been generated with Meson version {old_version}, '
-                         f'which is incompatible with the current version {current_version}.')
+                         f'which is incompatible with the current version {current_version}. '
+                         f'Consider reconfiguring the directory with meson setup --reconfigure')
```

Comment:
Don't you need `--wipe` in that case?

Reply:
I believe a `ninja reconfigure` was enough  when I got this last week, but I'm also not an expert here.

Reply:
@drawoc Could you tell me how to reproduce this? I tried, but failed.

Reply:
@Volker-Weissmann check out Meson 1.1.0, setup a build dir, checkout Meson 1.2.0, touch a meson.build file, run ninja, and the message should appear.

Reply:
This is what happens to me:
1. Build [mesa](https://gitlab.freedesktop.org/mesa/mesa/-/tree/main) with meson.
2. Run `meson devenv` to get an environment with the drivers available, and observe that everything works.
3. Get a meson upgrade from my package manager.
4. Run `meson devenv`, and now get the dreaded `Build directory has been generated with Meson version {old_version}, which is incompatible with the current version {current_version}`. Rebuilding does not help.

Reply:
I believe the most recent version update that caused this for me is:
```
[2023-08-07T18:02:54-0400] [ALPM] upgraded meson (1.1.1-1 -> 1.2.0-3)
```
using the packages from arch linux.

Reply:
Looking at my command history, I believe `ninja reconfigure` is what fixed it. I have no record of ever using the `--wipe` option.

Reply:
`ninja reconfigure is enough:
```
rm -rf ~/meson/12091/build ; git checkout 1.1.0 && ./meson.py ~/meson/12091 ~/meson/12091/build && git checkout 1.2.0 && ./meson.py devenv -C ~/meson/12091/build
```
fails, while
```
rm -rf ~/meson/12091/build ; git checkout 1.1.0 && ./meson.py ~/meson/12091 ~/meson/12091/build && git checkout 1.2.0 &&  ninja reconfigure -C ~/meson/12091/build && ./meson.py devenv -C ~/meson/12091/build
```
works.

Reply:
> @Volker-Weissmann check out Meson 1.1.0, setup a build dir, checkout Meson 1.2.0, touch a meson.build file, run ninja, and the message should appear.

No. This here:
```
rm -rf ~/meson/12091/build ; git checkout 1.1.0 && ./meson.py ~/meson/12091 ~/meson/12091/build && git checkout 1.2.0 && touch ~/meson/12091/meson.build && ninja -C ~/meson/12091/build
```
prints
```
Regenerating configuration from scratch: Build directory has been generated with Meson version 1.1.0, which is incompatible with the current version 1.2.0.
```
and works just fine.